dev_guide/event_notifications.adoc

```diff
@@ -0,0 +1,155 @@
+[[dev-guide-event-notifications]]
+= Event Notifications
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+For each of your projects, you can choose to receive email notifications about
+various xref:../dev_guide/events.adoc#dev-guide-events[events], including:
+
+* _Dead deployment_: A failed deployment in which no other deployment in the
+rollout history is able to remain active. The related `deploymentConfig` has no
+deployments with active replicas.
+
+* _Failed deployment_: A new deployment fails and another deployment in the
+ rollout history is now active. The related `deploymentConfig` has one active
+ deployment, but it is not the latest deployment.
+
+* _Dead build_: Occurs when automatically triggered builds are failing for a
+specified `buildConfig` that previously had successful builds.
+
+* _Dead persistentVolumeClaim_: A claim that is in the pending state for longer
+than some reasonable amount of time. These are claims whose resource requests
+likely do not match with any existing persistent volumes in the cluster.
+
+* _Lost persistentVolumeClaim_: A claim that is bound to a backing persistent
+volume, but the persistent volume has unexpectedly gone away.
+
+Notifications are sent to the email address of the creator of the project.
+
+[[event-notifications-how-it-works]]
+== How It Works
+
+A single thread analyzes the state of all objects and several object types in the
+cluster at the period defined by the resync interval. If a defined failure
+scenario is found, a notification object is created and added to the store with
+a specified date and time that the user should be notified. If a failure
+scenario is found and a notification for that scenario exists in the store, and
+its *notify at date/time* value is exceeded, the notification is added to the
+queue.
+
+Several worker threads watch the queue for notifications ready to be sent. Once
+the worker thread receives a notification from the queue, the notification is
+passed to the specified provider plug-in's send method. The provider plug-in
+handles sending the actual notification message. If the send was successful, the
+notification is marked as sent. The notification object remains in the store and
+its expiration date is updated. After the expiration period is passed, the
+notification object is deleted from the store and the failure scenario will
+cause a new notification object to be created. The user cannot be notified twice
+within the same expiration period.
+
+If a notification object exists in the store and the failure scenario is
+remedied, the notification object is deleted. If the object a notification
+was created for is deleted, the notification will eventually expire, be
+deleted, and never sent.
+
+[[event-notifications-building-from-source]]
```

Comment:
As discussed briefly on IRC, I suggest commenting out this section, as it's only relevant to OpenShift Online developers & operators. This applies to the following sections:

* event-notifications-building-from-source (only Online devs)
* deploying-on-a-cluster (only Online ops)
* event-notifications-adding-new-dependencies (Online devs)

This is good content to have, because it'll be needed once we merge this into origin. However, as `openshift/online` is a private repo, nobody will be able to do this, and users will not have access to `openshift-infra` (only our ops team has cluster admin)